src/WebJobs.Script/OutOfProc/Rpc/LanguageWorkerChannel.cs

```diff
@@ -343,6 +348,11 @@ internal void InvokeResponse(InvocationResponse invokeResponse)
                     context.ResultSource.TrySetException(responseEx);
                 }
             }
+
+            if ((_disposing || _disposed) && (!_disposeBufferHandle.SafeWaitHandle.IsClosed && _functionInputBuffers.Values.Sum(val => val.Count) < 1))
+            {
```

Comment:
@fabiocav - After looking through few options on how to address this issue, current iteration proposed by @yojagad uses waithandles. This LGTM! Can you please take a look as well to ensure we are following the right patterns here.